rackcity/views/rack_views.py

```diff
@@ -26,3 +30,66 @@ def rack_detail(request, pk):
     if request.method == 'GET':
         serializer = RackSerializer(rack)
         return JsonResponse(serializer.data)
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def rack_get(request):
+    """
+    List all racks within specified range.
+    """
+    range_serializer = RackRangeSerializer(data=request.data)
+    if range_serializer.is_valid():
+        try:
+            range_serializer.validate(request.data)
```

Comment:
Not the most beautiful order of of calls. But Django requires check on `.is_valid()` before any other methods can be called on the serializer object